### ISSUE TITLE
DNS settings aren't restored after suspend on Linux

### ISSUE THREAD

**Issue by cespare** [tailscale/tailscale#3473]
### What is the issue?

I'm trying out tailscale to see about using it at my company. Pretty good experience so far. Here's the first issue I've encountered (surprise! it's DNS):

On an Ubuntu host, after waking up from suspend, DNS settings are not restored for the tailscale interface. Wireguard is working fine (I can talk to other nodes by IP address); it's just DNS that is broken.

Basically it sounds the same as #2458, except that one was supposed to have been fixed.

I'm on Ubuntu 18.04. Systemd version is 237. (Yes, I need to update this machine.)

After waking up from suspend (when DNS isn't working):

```
$ systemd-resolve --status tailscale0
Link 10 (tailscale0)
      Current Scopes: none
       LLMNR setting: no
MulticastDNS setting: no
      DNSSEC setting: no
    DNSSEC supported: no
```

After kicking it via `sudo tailscale down; sudo tailscale up`:

```
$ systemd-resolve --status tailscale0
Link 10 (tailscale0)
      Current Scopes: DNS
       LLMNR setting: no
MulticastDNS setting: no
      DNSSEC setting: no
    DNSSEC supported: no
         DNS Servers: 172.20.0.2
          DNS Domain: liftoff.io
```

<details>
  <summary>Logs:</summary>
  
```
----------------------------------- after waking from suspend -----------------------------------------
Dec 01 21:50:25 eru tailscaled[30516]: [RATELIMIT] format("monitor: %s: src=%v, dst=%v, gw=%v, outif=%v, table=%v") (9 dropped)
Dec 01 21:50:25 eru tailscaled[30516]: monitor: RTM_DELROUTE: src=, dst=, gw=192.168.1.1, outif=2, table=254
Dec 01 21:50:25 eru tailscaled[30516]: monitor: RTM_DELROUTE: src=, dst=169.254.0.0/16, gw=, outif=2, table=254
Dec 01 21:50:25 eru tailscaled[30516]: monitor: RTM_DELROUTE: src=192.168.1.254/0, dst=192.168.1.0/24, gw=, outif=2, table=254
Dec 01 21:50:25 eru tailscaled[30516]: monitor: RTM_DELROUTE: src=, dst=fe80::/64, gw=, outif=2, table=254
Dec 01 21:50:25 eru tailscaled[30516]: [RATELIMIT] format("monitor: %s: src=%v, dst=%v, gw=%v, outif=%v, table=%v")
Dec 01 21:50:25 eru tailscaled[30516]: monitor: time jumped (probably wake from sleep); synthesizing major change event
Dec 01 21:50:25 eru tailscaled[30516]: control: setPaused(true)
Dec 01 21:50:25 eru tailscaled[30516]: LinkChange: all links down; pausing: interfaces.State{defaultRoute= ifs={tailscale0:[100.123.106.4/32 fd7a:115c:a1e0:ab12:4843:cd96:627b:6a04/128]} v4=false v6=false}
Dec 01 21:50:25 eru tailscaled[30516]: magicsock: SetNetworkUp(false)
Dec 01 21:50:25 eru tailscaled[30516]: magicsock: closing connection to derp-10 (network-down), age 3m0s
Dec 01 21:50:25 eru tailscaled[30516]: magicsock: 0 active derp conns
Dec 01 21:50:25 eru tailscaled[30516]: dns: Set: {DefaultResolvers:[] Routes:{liftoff.io.:[172.20.0.2:53]} SearchDomains:[liftoff.io.] Hosts:3}
Dec 01 21:50:25 eru tailscaled[30516]: dns: Resolvercfg: {Routes:{} Hosts:3 LocalDomains:[]}
Dec 01 21:50:25 eru tailscaled[30516]: control: mapRoutine: paused
Dec 01 21:50:25 eru tailscaled[30516]: control: mapRoutine: awaiting unpause
Dec 01 21:50:25 eru tailscaled[30516]: dns: OScfg: {Nameservers:[172.20.0.2] SearchDomains:[liftoff.io.] MatchDomains:[liftoff.io.]}
Dec 01 21:50:25 eru tailscaled[30516]: health("overall"): error: network down
Dec 01 21:50:25 eru tailscaled[30516]: wgengine: set DNS config again after major link change
Dec 01 21:50:25 eru tailscaled[30516]: magicsock: endpoints changed: 127.0.0.1:41641 (local), [::1]:41641 (local)
Dec 01 21:50:25 eru tailscaled[30516]: control: client.newEndpoints(0, [127.0.0.1:41641 [::1]:41641])
Dec 01 21:50:25 eru tailscaled[30516]: LinkChange: all links down; pausing: interfaces.State{defaultRoute= ifs={tailscale0:[100.123.106.4/32 fd7a:115c:a1e0:ab12:4843:cd96:627b:6a04/128]} v4=false v6=false}
Dec 01 21:50:29 eru tailscaled[30516]: LinkChange: all links down; pausing: interfaces.State{defaultRoute= ifs={tailscale0:[100.123.106.4/32 fd7a:115c:a1e0:ab12:4843:cd96:627b:6a04/128]} v4=false v6=false}
Dec 01 21:50:29 eru tailscaled[30516]: LinkChange: all links down; pausing: interfaces.State{defaultRoute= ifs={tailscale0:[100.123.106.4/32 fd7a:115c:a1e0:ab12:4843:cd96:627b:6a04/128]} v4=false v6=false}
Dec 01 21:50:30 eru tailscaled[30516]: logtail: dial "log.tailscale.io:443" failed: dial tcp: lookup log.tailscale.io on 127.0.0.53:53: server misbehaving (in 5.005s), trying bootstrap...
Dec 01 21:50:30 eru tailscaled[30516]: trying bootstrapDNS("derp9c.tailscale.com", "155.138.243.219") for "log.tailscale.io" ...
Dec 01 21:50:30 eru tailscaled[30516]: bootstrapDNS("derp9c.tailscale.com", "155.138.243.219") for "log.tailscale.io" error: Get "https://derp9c.tailscale.com/bootstrap-dns?q=log.tailscale.io": dial tcp 155.138.243.219:443: connect: network is unreachable
Dec 01 21:50:30 eru tailscaled[30516]: trying bootstrapDNS("derp8d.tailscale.com", "2a03:b0c0:1:d0::e08:e001") for "log.tailscale.io" ...
Dec 01 21:50:30 eru tailscaled[30516]: bootstrapDNS("derp8d.tailscale.com", "2a03:b0c0:1:d0::e08:e001") for "log.tailscale.io" error: Get "https://derp8d.tailscale.com/bootstrap-dns?q=log.tailscale.io": dial tcp [2a03:b0c0:1:d0::e08:e001]:443: connect: network is unreachable
Dec 01 21:50:30 eru tailscaled[30516]: trying bootstrapDNS("derp9.tailscale.com", "207.148.3.137") for "log.tailscale.io" ...
Dec 01 21:50:30 eru tailscaled[30516]: bootstrapDNS("derp9.tailscale.com", "207.148.3.137") for "log.tailscale.io" error: Get "https://derp9.tailscale.com/bootstrap-dns?q=log.tailscale.io": dial tcp 207.148.3.137:443: connect: network is unreachable
Dec 01 21:50:30 eru tailscaled[30516]: trying bootstrapDNS("derp12b.tailscale.com", "2001:19f0:5c01:48a:5400:3ff:fe8d:cb5f") for "log.tailscale.io" ...
Dec 01 21:50:30 eru tailscaled[30516]: bootstrapDNS("derp12b.tailscale.com", "2001:19f0:5c01:48a:5400:3ff:fe8d:cb5f") for "log.tailscale.io" error: Get "https://derp12b.tailscale.com/bootstrap-dns?q=log.tailscale.io": dial tcp [2001:19f0:5c01:48a:5400:3ff:fe8d:cb5f]:443: connect: network is unreachable
Dec 01 21:50:30 eru tailscaled[30516]: trying bootstrapDNS("derp3.tailscale.com", "68.183.179.66") for "log.tailscale.io" ...
Dec 01 21:50:30 eru tailscaled[30516]: bootstrapDNS("derp3.tailscale.com", "68.183.179.66") for "log.tailscale.io" error: Get "https://derp3.tailscale.com/bootstrap-dns?q=log.tailscale.io": dial tcp 68.183.179.66:443: connect: network is unreachable
Dec 01 21:50:30 eru tailscaled[30516]: trying bootstrapDNS("derp12.tailscale.com", "2001:19f0:5c01:289:5400:3ff:fe8d:cb5e") for "log.tailscale.io" ...
Dec 01 21:50:30 eru tailscaled[30516]: bootstrapDNS("derp12.tailscale.com", "2001:19f0:5c01:289:5400:3ff:fe8d:cb5e") for "log.tailscale.io" error: Get "https://derp12.tailscale.com/bootstrap-dns?q=log.tailscale.io": dial tcp [2001:19f0:5c01:289:5400:3ff:fe8d:cb5e]:443: connect: network is unreachable
Dec 01 21:50:30 eru tailscaled[30516]: trying bootstrapDNS("derp10.tailscale.com", "137.220.36.168") for "log.tailscale.io" ...
Dec 01 21:50:30 eru tailscaled[30516]: bootstrapDNS("derp10.tailscale.com", "137.220.36.168") for "log.tailscale.io" error: Get "https://derp10.tailscale.com/bootstrap-dns?q=log.tailscale.io": dial tcp 137.220.36.168:443: connect: network is unreachable
Dec 01 21:50:30 eru tailscaled[30516]: trying bootstrapDNS("derp2.tailscale.com", "2604:a880:2:d1::c5:7001") for "log.tailscale.io" ...
Dec 01 21:50:30 eru tailscaled[30516]: bootstrapDNS("derp2.tailscale.com", "2604:a880:2:d1::c5:7001") for "log.tailscale.io" error: Get "https://derp2.tailscale.com/bootstrap-dns?q=log.tailscale.io": dial tcp [2604:a880:2:d1::c5:7001]:443: connect: network is unreachable
Dec 01 21:50:30 eru tailscaled[30516]: trying bootstrapDNS("derp7.tailscale.com", "167.179.89.145") for "log.tailscale.io" ...
Dec 01 21:50:30 eru tailscaled[30516]: bootstrapDNS("derp7.tailscale.com", "167.179.89.145") for "log.tailscale.io" error: Get "https://derp7.tailscale.com/bootstrap-dns?q=log.tailscale.io": dial tcp 167.179.89.145:443: connect: network is unreachable
Dec 01 21:50:30 eru tailscaled[30516]: trying bootstrapDNS("derp12b.tailscale.com", "2001:19f0:5c01:48a:5400:3ff:fe8d:cb5f") for "log.tailscale.io" ...
Dec 01 21:50:30 eru tailscaled[30516]: bootstrapDNS("derp12b.tailscale.com", "2001:19f0:5c01:48a:5400:3ff:fe8d:cb5f") for "log.tailscale.io" error: Get "https://derp12b.tailscale.com/bootstrap-dns?q=log.tailscale.io": dial tcp [2001:19f0:5c01:48a:5400:3ff:fe8d:cb5f]:443: connect: network is unreachable
Dec 01 21:50:30 eru tailscaled[30516]: trying bootstrapDNS("derp4d.tailscale.com", "134.122.94.167") for "log.tailscale.io" ...
Dec 01 21:50:30 eru tailscaled[30516]: bootstrapDNS("derp4d.tailscale.com", "134.122.94.167") for "log.tailscale.io" error: Get "https://derp4d.tailscale.com/bootstrap-dns?q=log.tailscale.io": dial tcp 134.122.94.167:443: connect: network is unreachable
Dec 01 21:50:30 eru tailscaled[30516]: trying bootstrapDNS("derp2b.tailscale.com", "2604:a880:4:1d0::29:9000") for "log.tailscale.io" ...
Dec 01 21:50:30 eru tailscaled[30516]: bootstrapDNS("derp2b.tailscale.com", "2604:a880:4:1d0::29:9000") for "log.tailscale.io" error: Get "https://derp2b.tailscale.com/bootstrap-dns?q=log.tailscale.io": dial tcp [2604:a880:4:1d0::29:9000]:443: connect: network is unreachable
Dec 01 21:50:30 eru tailscaled[30516]: logtail: internet down; waiting
Dec 01 21:50:30 eru tailscaled[30516]: LinkChange: all links down; pausing: interfaces.State{defaultRoute= ifs={tailscale0:[100.123.106.4/32 fd7a:115c:a1e0:ab12:4843:cd96:627b:6a04/128]} v4=false v6=false}
Dec 01 21:50:30 eru tailscaled[30516]: LinkChange: all links down; pausing: interfaces.State{defaultRoute= ifs={tailscale0:[100.123.106.4/32 fd7a:115c:a1e0:ab12:4843:cd96:627b:6a04/128]} v4=false v6=false}
Dec 01 21:50:30 eru tailscaled[30516]: [RATELIMIT] format("LinkChange: all links down; pausing: %v")
Dec 01 21:50:32 eru tailscaled[30516]: logtail: internet back up
Dec 01 21:50:32 eru tailscaled[30516]: [RATELIMIT] format("monitor: %s: src=%v, dst=%v, gw=%v, outif=%v, table=%v") (15 dropped)
Dec 01 21:50:32 eru tailscaled[30516]: monitor: RTM_NEWROUTE: src=192.168.1.254/0, dst=192.168.1.254/32, gw=, outif=2, table=255
Dec 01 21:50:32 eru tailscaled[30516]: monitor: RTM_NEWROUTE: src=192.168.1.254/0, dst=192.168.1.255/32, gw=, outif=2, table=255
Dec 01 21:50:32 eru tailscaled[30516]: [RATELIMIT] format("monitor: %s: src=%v, dst=%v, gw=%v, outif=%v, table=%v")
Dec 01 21:50:32 eru tailscaled[30516]: control: setPaused(false)
Dec 01 21:50:32 eru tailscaled[30516]: LinkChange: major, rebinding. New state: interfaces.State{defaultRoute=enp0s31f6 ifs={enp0s31f6:[192.168.1.254/24] tailscale0:[100.123.106.4/32 fd7a:115c:a1e0:ab12:4843:cd96:627b:6a04/128]} v4=true v6=false}
Dec 01 21:50:32 eru tailscaled[30516]: magicsock: SetNetworkUp(true)
Dec 01 21:50:32 eru tailscaled[30516]: dns: Set: {DefaultResolvers:[] Routes:{liftoff.io.:[172.20.0.2:53]} SearchDomains:[liftoff.io.] Hosts:3}
Dec 01 21:50:32 eru tailscaled[30516]: dns: Resolvercfg: {Routes:{} Hosts:3 LocalDomains:[]}
Dec 01 21:50:32 eru tailscaled[30516]: control: mapRoutine: unpaused
Dec 01 21:50:32 eru tailscaled[30516]: control: mapRoutine: state:authenticated
Dec 01 21:50:32 eru tailscaled[30516]: magicsock: adding connection to derp-10 for home-keep-alive
Dec 01 21:50:32 eru tailscaled[30516]: magicsock: 1 active derp conns: derp-10=cr0s,wr0s
Dec 01 21:50:32 eru tailscaled[30516]: derphttp.Client.Recv: connecting to derp-10 (sea)
Dec 01 21:50:32 eru tailscaled[30516]: dns: OScfg: {Nameservers:[172.20.0.2] SearchDomains:[liftoff.io.] MatchDomains:[liftoff.io.]}
Dec 01 21:50:32 eru tailscaled[30516]: magicsock: adding connection to derp-1 for [Dn3a4]
Dec 01 21:50:32 eru tailscaled[30516]: magicsock: 2 active derp conns: derp-1=cr0s,wr0s derp-10=cr6ms,wr6ms
Dec 01 21:50:32 eru tailscaled[30516]: derphttp.Client.Send: connecting to derp-1 (nyc)
Dec 01 21:50:32 eru tailscaled[30516]: magicsock: derp-10 connected; connGen=1
Dec 01 21:50:32 eru tailscaled[30516]: health("overall"): ok
Dec 01 21:50:32 eru tailscaled[30516]: wgengine: set DNS config again after major link change
Dec 01 21:50:32 eru tailscaled[30516]: magicsock: closing connection to derp-10 (rebind), age 0s
Dec 01 21:50:32 eru tailscaled[30516]: magicsock: closing connection to derp-1 (rebind), age 0s
Dec 01 21:50:32 eru tailscaled[30516]: magicsock: 0 active derp conns
Dec 01 21:50:32 eru tailscaled[30516]: magicsock: adding connection to derp-10 for home-keep-alive
Dec 01 21:50:32 eru tailscaled[30516]: magicsock: 1 active derp conns: derp-10=cr0s,wr0s
Dec 01 21:50:32 eru tailscaled[30516]: magicsock: derp.Send(127.3.3.40:1): derphttp.Client.Send connect to region 1 (nyc): context canceled: context canceled
Dec 01 21:50:32 eru tailscaled[30516]: derphttp.Client.Recv: connecting to derp-1 (nyc)
Dec 01 21:50:32 eru tailscaled[30516]: health("overall"): error: not connected to home DERP region 10
Dec 01 21:50:32 eru tailscaled[30516]: magicsock: derp.Send(127.3.3.40:1): derphttp.Client closed
Dec 01 21:50:32 eru tailscaled[30516]: derphttp.Client.Recv: connecting to derp-10 (sea)
Dec 01 21:50:32 eru tailscaled[30516]: magicsock: derp-10 connected; connGen=1
Dec 01 21:50:32 eru tailscaled[30516]: health("overall"): ok
Dec 01 21:50:32 eru tailscaled[30516]: logtail: dial "log.tailscale.io:443" failed: dial tcp: lookup log.tailscale.io on 127.0.0.53:53: server misbehaving (in 56ms), trying bootstrap...
Dec 01 21:50:32 eru tailscaled[30516]: logtail: bootstrap dial succeeded
Dec 01 21:50:32 eru tailscaled[30516]: magicsock: endpoints changed: 50.35.79.188:41641 (stun), 192.168.1.254:41641 (local)
Dec 01 21:50:32 eru tailscaled[30516]: control: client.newEndpoints(0, [50.35.79.188:41641 192.168.1.254:41641])
Dec 01 21:50:35 eru tailscaled[30516]: control: mapRoutine: state:authenticated
Dec 01 21:50:37 eru tailscaled[30516]: magicsock: adding connection to derp-1 for [Dn3a4]
Dec 01 21:50:37 eru tailscaled[30516]: magicsock: 2 active derp conns: derp-1=cr0s,wr0s derp-10=cr5s,wr5s
Dec 01 21:50:37 eru tailscaled[30516]: derphttp.Client.Send: connecting to derp-1 (nyc)
Dec 01 21:50:37 eru tailscaled[30516]: magicsock: derp-1 connected; connGen=1
Dec 01 21:50:44 eru tailscaled[30516]: magicsock: disco: node [Dn3a4] d:f1e64442032b6d07 now using 3.86.30.230:41641
--------------     sudo tailscale down; sudo tailscale up ---------------------------
Dec 01 21:51:15 eru tailscaled[30516]: EditPrefs: MaskedPrefs{WantRunning=false}
Dec 01 21:51:15 eru tailscaled[30516]: control: setPaused(true)
Dec 01 21:51:15 eru tailscaled[30516]: Switching ipn state Running -> Stopped (WantRunning=false, nm=true)
Dec 01 21:51:15 eru tailscaled[30516]: magicsock: SetPrivateKey called (zeroed)
Dec 01 21:51:15 eru tailscaled[30516]: magicsock: closing connection to derp-10 (zero-private-key), age 43s
Dec 01 21:51:15 eru tailscaled[30516]: magicsock: closing connection to derp-1 (zero-private-key), age 38s
Dec 01 21:51:15 eru tailscaled[30516]: magicsock: 0 active derp conns
Dec 01 21:51:15 eru tailscaled[30516]: wgengine: Reconfig: configuring userspace wireguard config (with 0/0 peers)
Dec 01 21:51:15 eru tailscaled[30516]: control: mapRoutine: paused
Dec 01 21:51:15 eru tailscaled[30516]: control: mapRoutine: awaiting unpause
Dec 01 21:51:15 eru tailscaled[30516]: health("overall"): error: state=Stopped, wantRunning=false
Dec 01 21:51:15 eru tailscaled[30516]: wgengine: Reconfig: configuring router
Dec 01 21:51:15 eru tailscaled[30516]: [RATELIMIT] format("monitor: %s: src=%v, dst=%v, gw=%v, outif=%v, table=%v") (4 dropped)
Dec 01 21:51:15 eru tailscaled[30516]: monitor: RTM_DELROUTE: src=, dst=fd7a:115c:a1e0::/48, gw=, outif=10, table=52
Dec 01 21:51:15 eru tailscaled[30516]: monitor: RTM_DELROUTE: src=, dst=172.20.0.0/16, gw=, outif=10, table=52
Dec 01 21:51:15 eru tailscaled[30516]: magicsock: ReSTUN("link-change-minor") ignored; stopped, no private key
Dec 01 21:51:15 eru tailscaled[30516]: monitor: RTM_DELROUTE: src=, dst=172.30.0.0/16, gw=, outif=10, table=52
Dec 01 21:51:15 eru tailscaled[30516]: monitor: RTM_DELROUTE: src=, dst=172.31.0.0/16, gw=, outif=10, table=52
Dec 01 21:51:15 eru tailscaled[30516]: monitor: RTM_DELROUTE: src=, dst=fd7a:115c:a1e0:ab12:4843:cd96:627b:6a04/128, gw=, outif=10, table=255
Dec 01 21:51:15 eru tailscaled[30516]: [RATELIMIT] format("monitor: %s: src=%v, dst=%v, gw=%v, outif=%v, table=%v")
Dec 01 21:51:15 eru tailscaled[30516]: wgengine: Reconfig: configuring DNS
Dec 01 21:51:15 eru tailscaled[30516]: dns: Set: {DefaultResolvers:[] Routes:{} SearchDomains:[] Hosts:0}
Dec 01 21:51:15 eru tailscaled[30516]: dns: Resolvercfg: {Routes:{} Hosts:0 LocalDomains:[]}
Dec 01 21:51:15 eru tailscaled[30516]: dns: OScfg: {Nameservers:[] SearchDomains:[] MatchDomains:[]}
Dec 01 21:51:15 eru tailscaled[30516]: ipnserver: conn70: connection from userid 0; root has access
Dec 01 21:51:15 eru tailscaled[30516]: EditPrefs: MaskedPrefs{WantRunning=true}
Dec 01 21:51:15 eru tailscaled[30516]: transitioning to running; doing Login...
Dec 01 21:51:15 eru tailscaled[30516]: control: client.Login(false, 0)
Dec 01 21:51:15 eru tailscaled[30516]: control: setPaused(false)
Dec 01 21:51:15 eru tailscaled[30516]: Switching ipn state Stopped -> Starting (WantRunning=true, nm=true)
Dec 01 21:51:15 eru tailscaled[30516]: control: mapRoutine: unpaused
Dec 01 21:51:15 eru tailscaled[30516]: control: authRoutine: state:authenticated; wantLoggedIn=true
Dec 01 21:51:15 eru tailscaled[30516]: control: direct.TryLogin(token=false, flags=0)
Dec 01 21:51:15 eru tailscaled[30516]: control: doLogin(regen=false, hasUrl=false)
Dec 01 21:51:15 eru tailscaled[30516]: magicsock: SetPrivateKey called (init)
Dec 01 21:51:15 eru tailscaled[30516]: magicsock: private key changed, reconnecting to home derp-10
Dec 01 21:51:15 eru tailscaled[30516]: control: RegisterReq: onode= node=[MoNVg] fup=false
Dec 01 21:51:15 eru tailscaled[30516]: control: mapRoutine: state:authenticating
Dec 01 21:51:15 eru tailscaled[30516]: magicsock: adding connection to derp-10 for home-keep-alive
Dec 01 21:51:15 eru tailscaled[30516]: magicsock: 1 active derp conns: derp-10=cr0s,wr0s
Dec 01 21:51:15 eru tailscaled[30516]: derphttp.Client.Recv: connecting to derp-10 (sea)
Dec 01 21:51:15 eru tailscaled[30516]: wgengine: Reconfig: configuring userspace wireguard config (with 1/2 peers)
Dec 01 21:51:15 eru tailscaled[30516]: wgengine: Reconfig: configuring router
Dec 01 21:51:15 eru tailscaled[30516]: magicsock: derp-10 connected; connGen=1
Dec 01 21:51:15 eru tailscaled[30516]: wgengine: Reconfig: configuring DNS
Dec 01 21:51:15 eru tailscaled[30516]: dns: Set: {DefaultResolvers:[] Routes:{liftoff.io.:[172.20.0.2:53]} SearchDomains:[liftoff.io.] Hosts:3}
Dec 01 21:51:15 eru tailscaled[30516]: dns: Resolvercfg: {Routes:{} Hosts:3 LocalDomains:[]}
Dec 01 21:51:15 eru tailscaled[30516]: dns: OScfg: {Nameservers:[172.20.0.2] SearchDomains:[liftoff.io.] MatchDomains:[liftoff.io.]}
Dec 01 21:51:15 eru tailscaled[30516]: peerapi: serving on http://100.123.106.4:39319
Dec 01 21:51:15 eru tailscaled[30516]: peerapi: serving on http://[fd7a:115c:a1e0:ab12:4843:cd96:627b:6a04]:39319
Dec 01 21:51:15 eru tailscaled[30516]: Switching ipn state Starting -> Running (WantRunning=true, nm=true)
Dec 01 21:51:15 eru tailscaled[30516]: health("overall"): ok
Dec 01 21:51:15 eru tailscaled[30516]: ipnserver: conn70: ReadMsg: read unix /run/tailscale/tailscaled.sock->@: read: connection reset by peer
Dec 01 21:51:16 eru tailscaled[30516]: control: RegisterReq: got response; nodeKeyExpired=false, machineAuthorized=true; authURL=false
Dec 01 21:51:16 eru tailscaled[30516]: control: No AuthURL
Dec 01 21:51:16 eru tailscaled[30516]: control: authRoutine: state:authenticated; goal=nil paused=false
Dec 01 21:51:16 eru tailscaled[30516]: control: mapRoutine: state:authenticated
Dec 01 21:51:46 eru tailscaled[30516]: magicsock: disco: node [Dn3a4] d:f1e64442032b6d07 now using 3.86.30.230:41641
```
</details>

### OS

Linux

### OS version

Ubuntu 18.04

### Tailscale version

1.18.0

### Bug report

BUG-3390b59d07038792f49b0322e40734ca322a3f9799960525a8736aed9b3913f9-20211202053717Z-00136f7cb7347a7f

**Comment by bradfitz** [comment]
@cespare, thanks for the report!

@danderson, it looks like it's doing (or trying to do?) all the right things (other than unrelated log spam #3474). All the `dns: ...` lines look right. See anything I don't?

**Comment by josharian** [comment]
Hi, @cespare! Nice to see you here. And yeah, it's always DNS.

**Comment by cespare** [comment]
Oh, I think this was self-inflicted, actually. I *think* what happened was that I forgot to disable the wg-quick systemd service that I had installed (I had only temporarily stopped it when I initially installed tailscale). So after suspend I guess that wg-quick was coming up at the same time as tailscale and messing with my DNS config. In any case, I disabled that service at some point and now I don't see this issue anymore, so I'm pretty sure that was the cause. Sorry for the noise.
